cosmwasm/register-p2p-tokens.ts

```diff
@@ -0,0 +1,219 @@
+    if (untracked.length === 1) {
+        printInfo(`Untracked token ${tokenData.tokenId} on ${untracked[0].axelarChainId}`);
+        return untracked[0].axelarChainId;
+    }
+
+    // Use ethereum as the origin chain if the token is deployed on any of the Ethereum chains.
+    const ethereumChains = ['ethereum', 'core-ethereum', 'ethereum-sepolia', 'core-ethereum-sepolia', 'eth-sepolia'];
+    const ethereumChain = tokenData.chains.find((chain) => ethereumChains.includes(chain.axelarChainId.toLowerCase()));
+    if (ethereumChain) {
+        return ethereumChain.axelarChainId;
+    }
+
+    // Use the first chain that shows up.
+    return tokenData.chains[0].axelarChainId;
+}
+
+async function getSupply(tokenAddress: string, rpc: string): Promise<string> {
+    const provider = getDefaultProvider(rpc);
+    const token = new Contract(tokenAddress, IInterchainToken.abi, provider);
+    const supply = await token.totalSupply();
+    return supply.toString();
+}
+
+async function forEachTokenInFile(
+    config: ConfigManager,
+    options,
+    processToken: (tokenData: SquidToken, tokenOnChain: SquidTokenData) => Promise<void>,
+) {
+    const { env, tokenIds, chains } = options;
+    const tokenIdsToProcess = new Set(tokenIds);
+    const chainsToProcess = new Set(chains);
+    const tokenInfoString = fs.readFileSync(`axelar-chains-config/info/tokens-p2p/tokens-${env}.json`, 'utf8');
+    const tokenInfo = JSON.parse(tokenInfoString) as SquidTokenInfoFile;
+    const interchainTokenServiceAddress = config.getContractConfig('InterchainTokenService').address;
+
+    if (!interchainTokenServiceAddress) {
+        throw new Error('InterchainTokenService contract address not found');
+    }
+
+    const promises = Object.values(tokenInfo.tokens)
+        .filter((tokenData: SquidToken) => (tokenIds ? tokenIdsToProcess.has(tokenData.tokenId) : true))
+        .flatMap((tokenData: SquidToken) => {
+            return tokenData.chains
+                .filter((chain: SquidTokenData) => {
+                    try {
+                        return (
+                            tokenData.tokenType === 'interchain' &&
+                            (chains ? chainsToProcess.has(chain.axelarChainId.toLowerCase()) : true) &&
+                            chain.axelarChainId !== tokenData.originAxelarChainId &&
+                            (chain.registered ? !chain.registered : true) &&
+                            isConsensusChain(config.getChainConfig(chain.axelarChainId.toLowerCase()))
+                        );
+                    } catch (e) {
+                        printError(`Error getting chain config for ${chain.axelarChainId} (skipping chain): ${e}`);
+                        return false;
+                    }
+                })
+                .map(async (tokenOnChain: SquidTokenData) => {
+                    return processToken(tokenData, tokenOnChain);
+                });
+        });
+    await Promise.all(promises);
```

Comment:
This can't be done parallely. Txs from the same account need to be submitted sequentially due to sequence numbers